modules/dnn/src/caffe/caffe_importer.cpp

```diff
@@ -393,6 +393,7 @@ class CaffeImporter
             String name = layer.name();
             String type = layer.type();
             LayerParams layerParams;
+            layerParams.set("framework", "caffe");
```

Comment:
Can we avoid this flag?

Reply:
Agree.
Lets don't push information about frameworks on "execution" level. Let importers only to handle framework specifics.

If it is necessary, please add another flag which describes behavior instead of framework.

P.S. Especially we don't want to handle framework versions during the execution.